### PR TITLE
dev-8049 d3 bar chart setup

### PR DESCRIPTION
**Pull request by afnazu**
**High level description:**

Status of Funds chart - Set up d3 scale, axis, space for the labels, mobile layout, and tick marks before drawing horizontal bars.

**Technical details:**

Using mock data - API integration in future ticket

**JIRA Ticket:**
[DEV-8049](https://federal-spending-transparency.atlassian.net/browse/DEV-8049)

**Mockup:**
https://bahdigital.invisionapp.com/share/BSIAHSLA9PE#/screens/296067034

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [n/a] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [n/a] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [n/a] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [x] Design review complete `if applicable`
- [n/a] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete

### CODE REVIEW

File: src/js/components/agencyV2/visualizations/StatusOfFundsChart.jsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    const svg = d3.select('#sof_chart')
+        .append('svg')
+        .attr('class', 'svg')
+        .attr("viewBox", [0, 0, viewWidth + margins.left + margins.right, viewHeight + margins.top + margins.bottom])
+        .append('g')
+        .attr('transform', `translate(${isMobile ? margins.left - 40 : margins.left}, ${margins.top})`);
+    // scale to x and y data points
+    x.domain([0, sortedNums[0]]);
+    y.domain(resultNames);
+    const tickNum = isMobile ? 3 : 5;
```
Comment (ebdabbs) [inline src/js/components/agencyV2/visualizations/StatusOfFundsChart.jsx:110]: I'm not seeing it switch to 3 ticks in mobile, but that could just be the mock data!

**Review by ebdabbs — COMMENTED** [review]
Looking good! I'm wondering if there are any unit tests we could add to ensure we got the correct maximum or that the amounts are formatted correctly.

File: src/js/components/agencyV2/visualizations/StatusOfFundsChart.jsx (hunk 1 above)
Comment (afnazu) [reply src/js/components/agencyV2/visualizations/StatusOfFundsChart.jsx]: Updated to address all comments!

**Review by ebdabbs — APPROVED** [review]

File: src/js/components/agencyV2/statusOfFunds/VisualizationSection.jsx [hunk 2]
```diff
@@ -6,18 +6,38 @@
...
-        DEV-8049 horizontal bar chart viewing {levels[level]}s
-    </>
+const VisualizationSection = ({
+    level,
+    agencyName,
+    fy,
+    data
+}) => (
+    <div className="status-of-funds__visualization">
+        <h6>{agencyName} by <strong>{levels[level]}</strong> for FY {fy}</h6>
```
Comment (ebdabbs) [inline src/js/components/agencyV2/statusOfFunds/VisualizationSection.jsx:28]: Eventually `agencyName` should be updated to display either the agency name or the Sub-Component name (from Redux) depending on the level. We can do this as part of the click-through functionality ticket (DEV-8052)!

**Review by ebdabbs — APPROVED** [review]

**Pull request merged**